### PR TITLE
Fix default vscode settings sync value

### PR DESCRIPTION
**Pull request by jeanp413**
## Description

Fix default vscode settings sync value

## How to test

1. ~~Bundle gitpod-desktop vsix from https://github.com/gitpod-io/openvscode-server/pull/337~~ Dowload this vsix file [gitpod-desktop-0.0.30.zip](https://github.com/gitpod-io/gitpod/files/8482016/gitpod-desktop-0.0.30.zip)
2. Install vsix in vscode desktop
3. Press <kbd>F1</kbd> and execute `Gitpod: Turn on Settings Sync` command
4. Close all windows
6. Reopen vscode and enable settings sync: Press `f1` and execute  `Settings Sync: Turn On...` command (make sure you have some local settings already modified)
7. When it prompts to resolve conflicts, choose manual
8. Observe diff editor for settings.json shows remote value as `{}` (empty object) instead of `{settings:""}`

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by akosyakov** [comment]
@jeanp413 It would be nice if you can provide vsix file, otherwise it takes time to verify.

**Comment by jeanp413** [comment]
Added the vsix file description

**Comment by geropl** [comment]
I'm not sure how to test this. I followed the steps, but I did not see a conflict (used to sync with GitHub...?).
I assume I skipped some prior steps to do?

File: components/server/src/code-sync/code-sync-service.ts [hunk 1]
```diff
@@ -185,7 +185,8 @@ export class CodeSyncService {
                     value = await this.getTheiaCodeSyncResource(req.user.id);
                     version = 5;
                 } else if (resourceKey === SyncResource.Settings) {
-                    const settings = await this.userStorageResourcesDB.get(req.user.id, userSettingsUri);
+                    let settings = await this.userStorageResourcesDB.get(req.user.id, userSettingsUri);
+                    settings = settings === "" ? "{}" : settings;
```
[the remark this answers is not in the document]
Comment (jeanp413) [reply components/server/src/code-sync/code-sync-service.ts:189]: yeah but wanted to be more explicit, also I hope to remove this theia stuff soon :smile:

**Comment by akosyakov** [comment]
Setting sync worked for me, but I did not see any resolve conflict windows and could not follow verification steps because of it.

**Comment by akosyakov** [comment]
Settings sync worked btw, but extensions although synced were not installed, also I see errors in setting sync output:
Screenshot 2022-04-14 at 09 39 51
Screenshot 2022-04-14 at 09 40 02

Here exported logs: [vscodeLogs.zip](https://github.com/gitpod-io/gitpod/files/8487230/vscodeLogs.zip)

btw we should generate better file names for logs, something like `${workspace-id}-vscode-web-logs.zip`

**Comment by jeanp413** [comment]
@akosyakov  I see you had already some data in your account try deleting your sync data first and then settings sync again in vscode desktop
About the error seems an issue with openvsx maybe you were just unlucky, could you try again
```
[2022-04-14 09:37:01.468] [window] [error] [Extension Host] Fetch failed while querying open-vsx FetchError: network timeout at: https://open-vsx.jp-oauth-test.staging.gitpod-dev.com/api/-/query
	at Timeout.<anonymous> (/ide/extensions/gitpod-web/dist/extension.js:2:666420)
```

**Comment by geropl** [comment]
I tested again, and see:
 - `{}` for settings.json
 - empty file for keybindings.json
 - `[]` for extensions.json
@jeanp413 Is that expected?
image

**Pull request merged**